playbooks/roles/edxapp/templates/edx/app/edxapp/worker.sh.j2

```diff
@@ -4,19 +4,31 @@
 
 {% set edxapp_venv_bin = edxapp_venv_dir + "/bin" %}
 source {{ edxapp_app_dir }}/edxapp_env
+{% set executable = edxapp_venv_bin + '/celery' %}
+
 {% if COMMON_ENABLE_NEWRELIC_APP %}
-{% set executable = edxapp_venv_bin + '/newrelic-admin run-program ' + edxapp_venv_bin + '/celery' %}
+{% set executable = edxapp_venv_bin + '/newrelic-admin run-program ' + executable %}
 
 export NEW_RELIC_CONFIG_FILE="{{ edxapp_app_dir }}/newrelic.ini"
 if command -v ec2metadata >/dev/null 2>&1; then
   INSTANCEID=$(ec2metadata --instance-id);
   HOSTNAME=$(hostname)
   export NEW_RELIC_PROCESS_HOST_DISPLAY_NAME="$HOSTNAME-$INSTANCEID"
 fi
-{% else %}
-{% set executable = edxapp_venv_bin + '/celery' %}
 {% endif %}
 
+{% if EDXAPP_DATADOG_ENABLE %}
+{% set executable = edxapp_venv_bin + '/ddtrace-run ' + executable %}
+export DD_TAGS=("service:edx-edxapp-workers-{{ SERVICE_VARIANT }}", "queue:{{QUEUE_NAME}}")
```

Comment:
One last thing -- I think we can't use array syntax here, and just need a comma-delimited string:

```suggestion
export DD_TAGS="service:edx-edxapp-workers-{{ SERVICE_VARIANT }},queue:{{QUEUE_NAME}}"
```